### PR TITLE
Fix EntityScriptingInterface::getChildrenIDs function

### PR DESCRIPTION
**Pull request by wayne-chen**
- fix EntityScriptingInterface::getChildrenIDs to get children of entities, overlays, or avatars by parent ID

https://highfidelity.manuscript.com/f/cases/edit/1934/EntityScriptingInterface-getChildrenIDs-doesn-t-respect-avatars

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9798/HighFidelity-Beta-PR13172-9269bd261965dee53c1f489fcdd0ec1267bd9beb.apk

**Comment by wayne-chen** [comment]
To test:
- launch interface
- In menu, select Settings->Advanced Menus
- In menu, select Settings->Developer Menus
- In menu, open Edit->Console
- run similar or exact JavaScript code as shown below to generate entities of avatars, overlays, and entities in the Console
```
function createEntity(description, position, parent) {
    var entity = Entities.addEntity({
            type: "Sphere",
            position: position,
            dimensions: Vec3.HALF,
            dynamic: true,
            collisionless: true,
            parentID: parent,
            lifetime: 300  // Delete after 5 minutes.
        });
    return entity;
}

function createChildren() {
    var position = Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, {x: 0, y: 2, z: -5}));
    var avatar = MyAvatar.sessionUUID;
    if (avatar === Uuid.NULL) {
        avatar = MyAvatar.SELF_ID;
    }
    var textToWrite = "Avatar UUID: " + avatar + "\n\n";
    var entity = createEntity("AvatarChild" + 0, Vec3.sum(position, { x: 0, y: -1, z: 0 }), avatar);
    textToWrite += "AvatarChild UUID: " + entity + "\n";
    var overlay = Overlays.addOverlay("cube", {
        position: Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, { x: 0, y: 0, z: -3 })),
        rotation: MyAvatar.orientation,
        dimensions: { x: 0.3, y: 0.3, z: 0.3 },
        solid: true
    });
    textToWrite += "\nOverlay UUID: " + overlay + "\n\n";
    entity = createEntity("OverlayChild" + 0, Vec3.sum(position, { x: 0, y: -1, z: 0 }), overlay);
    textToWrite += "OverlayChild UUID: " + entity + "\n";
    var rootEntity = Entities.addEntity({
            type: "Sphere",
            position: position,
            dimensions: Vec3.HALF,
            dynamic: true,
            collisionless: true,
            lifetime: 300  // Delete after 5 minutes.
    });
    textToWrite += "\nEntity UUID: " + rootEntity + "\n\n";
    entity = createEntity("EntityChild", Vec3.sum(position, { x: 0, y: -1, z: 0 }), rootEntity);
    textToWrite += "EntityChild UUID: " + entity + "\n";
    console.log(textToWrite);
}

createChildren();
```
- find UUIDs printed in the console
- In Console, run `console.log(JSON.stringify(Entities.getChildrenIDs(MyAvatar.sessionUUIID)));`. The list returned should include all of the AvatarChild UUIDs printed above.
- In Console, run `console.log(JSON.stringify(Entities.getChildrenIDs(<Overlay UUID>)));`.  The list returned should include all of the OverlayChild UUIDs printed above.
- In Console, run `console.log(JSON.stringify(Entities.getChildrenIDs(<Entity UUID>)));`.  The list returned should include all of the EntityChild UUIDs printed 

above.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9798/HighFidelity-Beta-PR13172-9269bd261965dee53c1f489fcdd0ec1267bd9beb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9798/HighFidelity-Beta-Interface-PR13172-9269bd261965dee53c1f489fcdd0ec1267bd9beb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9798/HighFidelity-Beta-PR13172-9269bd261965dee53c1f489fcdd0ec1267bd9beb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9798/HighFidelity-Beta-Interface-PR13172-9269bd261965dee53c1f489fcdd0ec1267bd9beb.exe

File: libraries/entities/src/EntityScriptingInterface.h [hunk 1]
```diff
@@ -1216,12 +1216,12 @@ public slots:
 
 
     /**jsdoc
-     * Get the IDs of entities, overlays, and avatars that are directly parented to an entity. To get all descendants of an 
-     * entity, recurse on the IDs returned by the function.
+     * Get the IDs of entities, overlays, and avatars that are directly parented to an entity, overlay, or avatar model. To get all descendants of an 
+     * entity; overlay; or avatar, recurse on the IDs returned by the function.
```
Comment (ctrlaltdavid) [inline libraries/entities/src/EntityScriptingInterface.h]: Should be "entity, overlay, or avatar" - i.e., ","s instead of ";"s.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9825/HighFidelity-Beta-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9825/HighFidelity-Beta-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9825/HighFidelity-Beta-Interface-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9825/HighFidelity-Beta-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9825/HighFidelity-Beta-Interface-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.exe

**Comment by Qliemillar** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10165/HighFidelity-Beta-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.apk

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10165/HighFidelity-Beta-Interface-PR13172-92e9a8e064058644efd50f25bb03692c5c003597.dmg

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
PASS @conklin94122 Child UUIDs are reported after entering the parent UUID into console command

**Pull request merged**